src/components/Contacts/ContactDetails/ContactDetailsHeader/ContactHeaderSection/ContactHeaderStatusSection.test.tsx

```diff
@@ -8,25 +8,187 @@ import {
 } from '../ContactDetailsHeader.generated';
 import { ContactHeaderStatusSection } from './ContactHeaderStatusSection';
 
-const contact = gqlMock<ContactDetailsHeaderFragment>(
-  ContactDetailsHeaderFragmentDoc,
-  { mocks: { status: StatusEnum.PartnerFinancial } },
-);
+const contactMock = (status: StatusEnum) => {
+  return gqlMock<ContactDetailsHeaderFragment>(
+    ContactDetailsHeaderFragmentDoc,
+    { mocks: { status } },
+  );
+};
 
 describe('ContactHeaderStatusSection', () => {
-  it('should show loading state', async () => {
+  it('should show loading state', () => {
     const { queryByText } = render(
       <ContactHeaderStatusSection loading={true} contact={undefined} />,
     );
 
     expect(queryByText('Partner - Financial')).toBeNull();
   });
 
-  it('should render with contact details', async () => {
+  it('should render if status is null', () => {
```

Comment:
You could use https://jestjs.io/docs/api#testeachtablename-fn-timeout for all of these. The syntax is a little weird but it would clean up the repetition. I thought I had used it a couple times but [this](https://github.com/CruGlobal/missionhub-react-native/blob/500b54489c88f117bba1354ff2b4ec312ec03d66/src/utils/hooks/__tests__/useRefreshing.ts) is the only instance I found.

I'm fine with how it is though 🤷‍♂️

Reply:
Ah, I like that. I'll see if I can get it working.

Reply:
Works great, thanks for the suggestion 🙂